pkg/adapter/v2/cloudevents.go

```diff
@@ -58,6 +58,7 @@ func newCloudEventsClientCRStatus(env EnvConfigAccessor, target string, ceOverri
 	pOpts = append(pOpts, cloudevents.WithRoundTripper(&ochttp.Transport{
 		Propagation: tracecontextb3.TraceContextEgress,
 	}))
+	pOpts = append(pOpts, http.WithIsRetriableFunc(retryIfGreaterThan300))
```

Comment:
This is a breaking change. PingSource is using the default values.

/hold

Reply:
I changed the default values because I need [eventshub](https://github.com/knative-sandbox/reconciler-test/blob/0071c48281c750978b954d2dea496942faa6259d/pkg/eventshub/receiver/receiver.go#L214) or [event-flaker](https://github.com/knative/eventing/blob/main/test/test_images/event-flaker/main.go#L62)  to simulate a events lost scenario for e2e tests. But when dropping events, the flaker return 500 and eventshub return 409, all these two status codes are not in the defaults retriable values for cloudevents.Client.  

I refer to the retry implementation of [eventing-rocketmq](https://github.com/knative-sandbox/eventing-rabbitmq/blob/main/pkg/dispatcher/dispatcher.go#L80),robbitmq retries when status bigger than 300 or less than 200.  and [message sender](https://github.com/knative/eventing/blob/main/pkg/kncloudevents/message_sender.go#L163-L166) , the sender reties when status bigger than 300

I am a little confused about which status codes the sink returns should be triggered a retry? and do we need to unify the retry rules for all components?

Reply:
How do you think we retry according to the implementation similar to [go-retryablehttp](https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L458)

Reply:
@lionelvillard  How about we retry according to [go-retryablehttp](https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L458) 's rule?  or only retry when the status code is 409? Could you give me some advice?

Reply:
I'd generally like to see the `go-retyablehttp` lib used

Reply:
@xinydev  some realted PR: https://github.com/knative/eventing/pull/5285
(lot's of discussion on there - it's for sure not an "easy" topic, :sweat_smile: )

Reply:
@cr22rc are you ok with this change?

`retryIfGreaterThan300` is temporary until  #5285 lands.

Reply:
My preference would be if these were not hard coded.  Any specific reason they can't be made to be adjusted by a config map or environment variable at the very least?

Reply:
Thanks for all the suggestions, I will withdraw this change, keep its default value, and then add an environment variable to configure the return http status code of [eventshub](https://github.com/knative-sandbox/reconciler-test/issues/211), which may be more reasonable.